cpp/common/src/codingstandards/cpp/Operator.qll

```diff
@@ -264,3 +267,51 @@ class UserOverloadedOperator extends Function {
     not this.isCompilerGenerated()
   }
 }
+
+/**
+ * A `std::basic_istream` class, or something that can be used
+ * as one. Based on the BasicOStreamClass.
+ */
+private class BasicIStreamClass extends Type {
```

Comment:
should this go in FileStreams.qll? and maybe extend the [pre-existing modelling](https://github.com/github/codeql-coding-standards/blob/main/cpp/common/src/codingstandards/cpp/standardlibrary/FileStreams.qll#L26)? (didnt look too closely at what the difference is, as in why the hierarchy on that is narrowed to class template instantiations only on the pre-existing model for what I think is maybe meant to represent the same thing - confirm on that!)